homeassistant/components/daikin/number.py

```diff
@@ -0,0 +1,146 @@
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from . import DOMAIN as DAIKIN_DOMAIN, DaikinApi
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up Daikin climate based on config_entry."""
+    daikin_api: DaikinApi = hass.data[DAIKIN_DOMAIN][entry.entry_id]
+    if zones := daikin_api.device.zones:
+        async_add_entities(
+            [
+                DaikinZoneTemperature(daikin_api, zone_id)
+                for zone_id, zone in enumerate(zones)
+                if zone[0] != "-" and zone[2] != 0
+            ]
+        )
+
+
+class DaikinZoneTemperature(NumberEntity):
+    """Representation of a zone temperature setting."""
+
+    _attr_icon = "mdi:thermostat"
+    _attr_native_step = 1
+    _attr_has_entity_name = True
+    _attr_device_class = NumberDeviceClass.TEMPERATURE
+    _attr_native_unit_of_measurement = UnitOfTemperature.CELSIUS
+    _registry: er.EntityRegistry | None
+    _remove_listener: Callable[[], None] | None
+    _target_temperature: float
+    _previous_value: float | None
+    _current_value: float
+
+    def __init__(self, daikin_api: DaikinApi, zone_id) -> None:
+        """Initialize the zone."""
+        self._api = daikin_api
+        self._zone_id = zone_id
+        self._registry = None
+        self._remove_listener = None
+        self._target_temperature = self._api.device.target_temperature
+        self._previous_value = None
+        self._current_value = self._api.device.zones[self._zone_id][2]
+        self._attr_device_info = self._api.device_info
+        self._attr_unique_id = f"{self._api.device.mac}-zone-temp{self._zone_id}"
+        self._attr_name = f"{self._api.device.zones[self._zone_id][0]} temperature"
+        self._attr_native_min_value = self._target_temperature - 2
+        self._attr_native_max_value = self._target_temperature + 2
+
+    @property
+    def native_value(self) -> float:
+        """Return the state of the sensor."""
+        if self.has_updated_value():
+            self._previous_value = None
+            self._current_value = self._api.device.zones[self._zone_id][2]
+        return self._current_value
+
+    def has_updated_value(self) -> bool:
+        """Detect if api has an updated value."""
+        if self._previous_value == self._api.device.zones[self._zone_id][2]:
+            return False
+        if self.value_in_range(self._api.device.zones[self._zone_id][2]):
+            return True
+        return False
+
+    def value_in_range(self, value: float) -> bool:
+        """Check temperature is within range."""
+        if value < (self._target_temperature - 2) or value > (
+            self._target_temperature + 2
+        ):
+            return False
+        return True
+
+    async def async_added_to_hass(self) -> None:
```

Comment:
Should not be needed

Reply:
There is no value in having a listener here when `async_update` will be called automatically

Reply:
I used a listener here to remove the delay in calling async_update. The delay was allowing the temperature to be set out of range until async_update is called.

Reply:
What delay? All logic should be within `async_update()` and you should now the min/max values or you need to fetch them before setting a temperature so you know it can be done or then raise if it's out of range.
Sounds like the min/max would be a floating target which also seems a bit odd?

Reply:
It is a floating target based on the target temperature of the climate entity, plus or minus 2 degrees. If I change the target temperature there is a delay in updating the min/max values.

Reply:
So you should probably implement a data update coordinator before this PR then which can coordinate all your data and refresh data more easily than in individual platforms.